### DISCUSSION TITLE
Enforcing/validating the input of a function to not be undefined

### DISCUSSION THREAD

**Discussion by ashley-hawkins** [napi-rs/napi-rs#2282] [Q&A] [answered]
If I define a function like so:

```rs
#[napi]
pub fn example(env: Env, callback: Function<(String), JsUndefined>) -> napi::Result<()> {
  callback.call("hi".to_owned())?;

  Ok(())
}
```

example() can be called without any arguments, and that causes a failure with the message "invalid arg", I guess because it's not possible to call `undefined`. But is it possible to check whether the value being received by my function as a parameter is valid in the first place? To me it seems like `undefined` shouldn't be an accepted input for a Function, am I missing something> Does anyone know if it's possible to validate it?

Edit: I've noticed that I can use `#[napi(strict)]` to have it enforce the argument types, but I'm still wondering if I just have a Function without strict, would I be able to check whether it's actually valid?

**Comment by Brooooooklyn** [comment, marked as answer]
You can use `Either`:
```rust
#[napi]
pub fn example(env: Env, maybe_callback: Either<Function<(String), ()>, Unknown>) -> napi::Result<()> {
  match maybe_callback {
    Either::A(callback) => {
      callback.call("hi".to_owned())?;
    }
    Either::B(whatever) => {},
  }
  Ok(())
}
```
